### PR TITLE
Last fixes

### LINKED ISSUE #64: Stars status is lost when searching notes

### LINKED ISSUE #65: Autofocus wrong username
In permission modal

### LINKED ISSUE #73: Add ability to star public note
It already works, it's just missing a button on the editor page

### LINKED ISSUE #74: Autosave shouldn't save if the user hasn't typed anything

### LINKED ISSUE #79: Add favicon

### PR DESCRIPTION
**Pull request by Blatoy**
This (WAY too big) PR adds the following : 
- Code cleanup of both editor and home page
- Stability improvement for logged out users
- Close #73 
- Fix #74 
- Close #65 
- Close #79 
- Fix #64 
- Fix scrollbar position when not enough notes are loaded
- Add a bit of responsivity to the website

### CODE REVIEW

File: marcdown_project/frontend/src/components/ModalPermissions.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+        query(`/api/note/${this.props.note.id}/`, "PUT", {
+            sharedWith: newSharedArray
+        }).then((result) => {
+            // Remove bad sharers and update
+            if (result.badSharerNames.length > 0) {
+                this.setState({ inputSharer: "User not found" });
+                document.getElementById("input-sharer").select();
+                newSharedArray.splice(newSharedArray.indexOf(result.badSharerNames), 1);
+            }
```
Comment (g-wizzy) [inline marcdown_project/frontend/src/components/ModalPermissions.js:62]: :+1:

File: marcdown_project/frontend/src/components/ModalPermissions.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        });
+    }
+
+    _removeSharedUser(index) {
+        let sharedWith = this.props.note.sharedWith;
+        sharedWith.splice(index, 1);
+
+        query(`/api/note/${this.props.note.id}/`, "PUT", {
+            sharedWith: sharedWith
+        });
```
Comment (g-wizzy) [inline marcdown_project/frontend/src/components/ModalPermissions.js:75]: Is this assuming all the other users in the list are correct ?

File: marcdown_project/frontend/src/components/Tag.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
 
 import React, { Component } from "react";
 
+/**
+ * Literally a span with a specific class, could probably be deleted
+ */
 export default class Tag extends Component {
```
Comment (g-wizzy) [inline marcdown_project/frontend/src/components/Tag.js:7]: Don't worry about it. Purely semantic components are perfectly fine.

**Review by g-wizzy — APPROVED** [review]

File: marcdown_project/frontend/src/components/ModalPermissions.js (hunk 2 above)
Comment (Blatoy) [reply marcdown_project/frontend/src/components/ModalPermissions.js:75]: Yes, it should not be possible to have invalid users in the list when you click on a user to delete it

**Pull request merged**